docs/docs/features/monitoring.md

```diff
@@ -0,0 +1,113 @@
+    # a named volume defined in the bottom of the Compose file; it can also be a mounted folder
+    - prometheus-data:/prometheus
+```
+
+You will also need to add `prometheus-data` to the list of volumes in the bottom of the Compose file:
+
+```yaml
+volumes:
+  model-cache:
+  prometheus-data:
+```
+
+The last piece is the [configuration file][prom-file]. This file defines (among other things) the sources Prometheus should target. Download it and place it in the same folder as the Compose file.
+
+:::tip
+The provided file is just a starting point. There are a ton of ways to configure Prometheus, so feel free to experiment!
+:::
+
+After bringing down the containers with `docker compose down` and back up with `docker compose up -d`, a Prometheus instance will now collect metrics from the immich server and microservices containers. Note that we didn't need to expose any new ports for these containers - the communication is handled in the internal Docker network.
+
+:::note
+To see exactly what metrics are made available, you can additionally add `8081:8081` to the server container's ports and `8082:8081` to the microservices container's ports. Visiting the `/metrics` endpoint for these services will show the same raw data that Prometheus collects.
+:::
+
+### Usage
+
+So after setting up Prometheus, how do you actually view the metrics? The simplest way is to use Prometheus directly. Visiting Prometheus will show you a web UI where you can search for and visualize metrics. You can also view the status of your data sources and configure settings, but this is beyond the scope of this guide.
+
+## Grafana
+
+For a dedicated tool with nice presentation, you can use Grafana instead. This connects to Prometheus (and possibly other source) for sophisticated data visualization.
+
+Setting up Grafana is similar to Prometheus. You can add a service for it:
+
+```yaml
+immich-grafana:
+  container_name: immich_grafana
+  command: ['./run.sh', '-disable-reporting'] # this is to disable Grafana's telemetry
+  ports:
+    - 3000:3000
+  image: grafana/grafana
+  volumes:
+    # stores your pretty dashboards and panels
+    - grafana-data:/var/lib/grafana
+```
+
+And add another volume for it:
+
+```yaml
+volumes:
+  model-cache:
+  prometheus-data:
+  grafana-data:
+```
+
+After bringing down the services and back up again, you can now visit Grafana to view your metrics. On the first login, enter `admin` for both username and password and update your password. You can then go to the settings and add a data source with `http://immich-prometheus:9090` to point Grafana to your Prometheus instance.
+
+### Usage
+
+You can make your first dashboard to get started. Don't forget to save it frequently, or you'll lose all your progress!
+
+You can then make a new panel, specifying Prometheus as the data source for it.
+
+-- TODO: add images and more details here
+
+[prom-file]: https://github.com/immich-app/immich/releases/latest/download/prometheus.yml
```

Comment:
The release workflow doesn't yet add this file to the artifacts.

Reply:
Done